### PR TITLE
Add metrics support to the OTLP exporter

### LINKED ISSUE #377: Add opentelemetry collector/agent exporter for metrics
Add exporter to export Metrics to Open-telemetry collector.
This is blocked on release of [open-telemetry/protocol](https://github.com/open-telemetry/opentelemetry-proto).

### PR DESCRIPTION
**Pull request by MrAlias**
Adds an `Export` method to the OTLP exporter to export metrics from OpenTelemetry instrumentation in the OTLP format.

Still a few things to clean up, but conceptually there. Things I'm hoping to clean up:

- [x] Add comments and docstrings.
- [x] More unit test.
- [x] Rebase the upstream changes into this.

Resolves #377

### CODE REVIEW

File: exporters/otlp/internal/transform/metric.go [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+		}
+	case core.Float64NumberKind:
+		m.MetricDescriptor.Type = metricpb.MetricDescriptor_COUNTER_DOUBLE
+		m.DoubleDatapoints = []*metricpb.DoubleDataPoint{
+			{Value: sum.CoerceToFloat64(n)},
+		}
+	}
+
+	return m, nil
+}
```
Comment (rghetia) [inline exporters/otlp/internal/transform/metric.go]: Record with same metric descriptor but different label values could be grouped.
```go
m1 {
   descriptor: { ..}
   datapoints: []{
      {
        Labels: { {"k1", "v11"}. {"k2", "v21"}},
        value: 50
       },
      {
        Labels: { {"k1", "v12"}. {"k2", "v22"}},
        value: 500
       },
     }
}
```
It is only beneficial if there are large number of time-series for the same metrics.

Anyway, It doesn't need to be part of this PR.

**Comment by rghetia** [comment]
@MrAlias , build failed after I rebased it. can you PTAL?

File: exporters/otlp/internal/transform/metric.go (hunk 1 above)
Comment (MrAlias) [reply exporters/otlp/internal/transform/metric.go]: Agreed. I was planning to address batching in the "Add support for Resources" work on this. I'll be sure to link this comment with that Issue to track it.

Comment (MrAlias) [reply exporters/otlp/internal/transform/metric.go]: I created #549 to track this and to break up #515 into dedicated work on implementing Resources in the SDK.

**Comment by jmacd** [comment]
:tada:

**Pull request merged**